.github/workflows/build.yml

```
@@ -12,31 +12,26 @@ jobs:
       matrix:
         # Use these Java versions
         java: [
-            17    # Minimum supported by Minecraft
+          21,    # Current Java LTS
         ]
-        # and run on both Linux and Windows
-        os: [ubuntu-20.04, windows-latest]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-22.04
     steps:
       - name: checkout repository
-        uses: actions/checkout@v2
-        with:
-            fetch-depth: 0
+        uses: actions/checkout@v4
```

Actually, this change broke the workflow - you need it like this :

```suggestion
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
          filter: tree:0
```
(that added filter is not strictly necessary, but it's an optimization made possible by checkout V4)